yearn/prices/magic.py

```diff
@@ -156,22 +67,39 @@ def get_price_eth(token, block=None):
         price = balancer.get_price(token, block=block)
         logger.debug("balancer pool -> %s", price)
 
-    # peel a layer from [multiplier, underlying]
+    elif yearn_lens.is_yearn_vault(token):
+        price = yearn_lens.get_price(token, block=block)
+        logger.debug("yearn -> %s", price)
+    # xcredit
+    elif token == '0xd9e28749e80D867d5d14217416BFf0e668C10645':
+        logger.debug('xcredit -> unwrap')
+        wrapper = contract(token)
+        price = get_price(wrapper.token(), block=block) * wrapper.getShareValue() / 1e18
+
+    markets = [
+        chainlink,
+        curve.curve,
+        compound,
+        fixed_forex,
+        synthetix,
+        band,
+        uniswap_v2,
+        uniswap_v3,
+        uniswap_v1
+    ]
+    for market in markets:
+        if price:
+            break
+        if market and token in market:
+            logger.debug("getting price for token %s with market %s", token, market)
```

Comment:
not super happy with using `__repr__` here, it has a bit different purpose. maybe change to a class-level `name` or even `market.__class__.__name__`.

Reply:
changed!